pkg/execute/executor.go

```diff
@@ -414,3 +494,32 @@ func showControllerConfig() (configYaml string, err error) {
 
 	return configYaml, nil
 }
+
+// authorizeCommandByProfile function check if channel has permission to run the specific command or not based on access rules defined in corresponding profile
+func authorizeCommandByProfile(Profile config.Profile, args []string) bool {
+	authorizedCommand := false
+
+	if len(args) >= 2 {
+		// check for allowed kubectl verb
+		allowedOperations := Profile.Kubectl.Commands.Verbs
+		if authorizedCommand = utils.Contains(allowedOperations, args[0]); !authorizedCommand {
+			return false
+		}
+
+		// check for allowed kubectl resource
+
+		AllowedKubectlResourceMap = make(map[string]bool)
+		for _, r := range Profile.Kubectl.Commands.Resources {
+			AllowedKubectlResourceMap[r] = true
+		}
```

Comment:
Iterating through the config for each command is costly. Can we have these maps as a part of profile/binding struct and initialize those while creating config structs?